src/transforms/remap.rs

```diff
@@ -133,14 +135,53 @@ pub struct RemapConfig {
     #[configurable(derived, metadata(docs::hidden))]
     #[serde(default)]
     pub runtime: VrlRuntime,
+
+    #[configurable(derived, metadata(docs::hidden))]
+    #[serde(skip)]
+    #[derivative(Debug = "ignore")]
+    /// Cache can't be `BTreeMap` or `HashMap` because of `TableRegistry`, which doesn't allow us to inspect tables inside it.
+    /// And even if we allowed the inspection, the tables can be huge, resulting in a long comparison or hash computation
+    /// while using `Vec` allows us to use just a shallow comparison
+    pub cache: Mutex<
+        Vec<(
+            (TableRegistry, schema::Definition),
+            std::result::Result<(Program, String, MeaningList), String>,
+        )>,
```

Comment:
Nit/TIOLI: Personal pet peeve, I dislike tuples with > 2 elements in them. We could use custom `type`s here to improve readability.


```suggestion
    type CacheKey = (TableRegistry, schema::Definition);
    type CacheValue = (Program, String, MeaningList);
    
    pub cache: Mutex<
        Vec<(CacheKey, std::result::Result<CacheValue, String>)>,
```